itools-packager/src/main/java/com/powsybl/itools/ItoolsPackagerMojo.java

```diff
@@ -79,33 +81,125 @@ public void setFiles(File[] files) {
+                            entry.setName(baseDir.relativize(file).toString());
+                            entry.setSize(file.toFile().length());
+                            entry.setMode(0770);
+                            tos.putArchiveEntry(entry);
+                            Files.copy(file, tos);
+                            tos.closeArchiveEntry();
+
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        @Override
+                        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                            tos.putArchiveEntry(new TarArchiveEntry(baseDir.relativize(dir).toString() + "/"));
+                            tos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
                 }
+                break;
+
+            case ".tar.bz2":
+                try (TarArchiveOutputStream tos = new TarArchiveOutputStream(new BZip2CompressorOutputStream(new BufferedOutputStream(fos)))) {
+                    tos.setBigNumberMode(TarArchiveOutputStream.BIGNUMBER_STAR); // to store big number in the archive.
+                    tos.setLongFileMode(TarArchiveOutputStream.LONGFILE_GNU); // to store long file names in the archive.
+
+                    Files.walkFileTree(dir, new SimpleFileVisitor<Path>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+                            TarArchiveEntry entry = new TarArchiveEntry(new File(file + "/"));
+                            entry.setName(baseDir.relativize(file).toString());
+                            entry.setSize(file.toFile().length());
+                            entry.setMode(0770);
+                            tos.putArchiveEntry(entry);
+                            Files.copy(file, tos);
+                            tos.closeArchiveEntry();
 
-                @Override
-                public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
-                    zos.putArchiveEntry(new ZipArchiveEntry(baseDir.relativize(dir).toString() + "/"));
-                    zos.closeArchiveEntry();
-                    return FileVisitResult.CONTINUE;
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        @Override
+                        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+                            tos.putArchiveEntry(new TarArchiveEntry(baseDir.relativize(dir).toString() + "/"));
+                            tos.closeArchiveEntry();
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
                 }
```

Comment:
This code is a copy/past of what you did for `tar.gz` except the type of the compressor (`GZIPOutputStream` vs `BzipCompressorOutputStream`). Should extract the `SimpleFileVisitor` anonymous class in a separate class file to share the code.

Reply:
I created a BrowseFileTree class file containing 2 methods createArchiveZip and createArchiveTar.
The createArchiveTar method is used to process 2 types of the compressor: GZIPOutputStream and BzipCompressorOutputStream.
However, I have not managed to merge two methods createArchiveZip and createArchiveTar that are similar too but I do not know if this is feasible in Java 8 or more?